neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/data/im/BoundInMemoryBackend.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2013-2017 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.data.im;
+
+import fr.inria.atlanmod.commons.annotation.Singleton;
+import fr.inria.atlanmod.commons.annotation.Static;
+import fr.inria.atlanmod.commons.log.Log;
+import fr.inria.atlanmod.neoemf.core.Id;
+import fr.inria.atlanmod.neoemf.data.bean.ClassBean;
+import fr.inria.atlanmod.neoemf.data.bean.FeatureBean;
+import fr.inria.atlanmod.neoemf.data.bean.SingleFeatureBean;
+import fr.inria.atlanmod.neoemf.data.mapping.DataMapper;
+
+import net.openhft.chronicle.map.ChronicleMap;
+import net.openhft.chronicle.map.ChronicleMapBuilder;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.annotation.Nonnegative;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import static fr.inria.atlanmod.commons.Preconditions.checkArgument;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+/**
+ * A {@link InMemoryBackend}, bound to a unique {@link Id}, that stores all elements in {@link Map}s.
+ */
+@ParametersAreNonnullByDefault
+public final class BoundInMemoryBackend extends AbstractInMemoryBackend {
+
+    /**
+     * The number of instances of this class.
+     *
+     * @see #innerClose()
+     * @see DataHolder#close(Id)
+     */
+    @Nonnull
+    @Nonnegative
+    // FIXME May be inconsistent if a PersistentEObject is released by the garbage collector
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)